Boolean.CSharp.Main/Core.cs

```diff
@@ -1,14 +1,123 @@
-﻿using System;
+﻿using Newtonsoft.Json;
+using System;
 using System.Collections.Generic;
 using System.Linq;
 using System.Text;
+using System.Transactions;
 
 
 namespace Boolean.CSharp.Main
 {
-    public class Core
+    public enum accountType
     {
+        current,
+        savings
+    }
+    public class Customer
+    {
+        private List<Account> accounts = new List<Account>();
+        public List<Account> Accounts { get { return accounts; } }
+        public Account createAccount(int accountNr, accountType type)
+        {
+            if(type == accountType.current)
+            {
+                CurrentAccount account = new CurrentAccount(accountNr);
+                accounts.Add(account);
+                return account;
+            }
+            else
+            {
+                SavingsAccount account = new SavingsAccount(accountNr);
+                accounts.Add(account);
+                return account;
+            }
+        }
+    }
+    public abstract class Account
+    {
+        private int _accountNr;
+        private float _balance = 0;
+        private List<Transaction> transaction = new List<Transaction>();
+        public List<Transaction> Transaction { get { return transaction; } }
+        public int AccountNr { get { return _accountNr; } }
+        public float Balance { get { return _balance; } }
+        public Account(int accountNr)
+        {
+            _accountNr = accountNr;
+        }
+        public Transaction withdraw(float amount)
+        {
+            _balance -= amount;
+            DateTime time = DateTime.Now;
+            Transaction tempT = new Transaction(amount, transactionType.withdraw, time, _balance);
+            Transaction.Add(tempT);
+            return tempT;
+        }
+        public Transaction deposit(float amount)
+        {
+            _balance += amount;
+            DateTime time = DateTime.Now;
+            Transaction tempT = new Transaction(amount, transactionType.deposit, time, _balance);
+            Transaction.Add(tempT);
+            return tempT;
+        }
+        public void printStatement()
+        {
+            Console.WriteLine("       date          || credit   ||   debit   ||  balance  ");
```

Comment:
print statement should not output to consoel, instead it should return your banka ccount statement as a string, or list of strings... there's also no way to test the console output...